opendistro-for-elasticsearch/plan.sh

```diff
@@ -0,0 +1,94 @@
+ELASTICSEARCH_VERSION="6.5.4"
+ELASTICSEARCH_PKG_URL="https://artifacts.elastic.co/downloads/elasticsearch/elasticsearch-oss-$ELASTICSEARCH_VERSION.tar.gz"
+pkg_version=0.7.0.1
+pkg_name="opendistro-for-elasticsearch"
+pkg_description="An Apache 2.0-licensed distribution of Elasticsearch enhanced with enterprise security, alerting, SQL, and more"
+pkg_origin="core"
+pkg_maintainer="The Habitat Maintainers <humans@habitat.sh>"
+pkg_license=("Apache-2.0")
+pkg_upstream_url="https://github.com/opendistro-for-elasticsearch"
+pkg_build_deps=(
+  core/git
+  core/maven
+  core/zip
+)
+pkg_deps=(
+  core/coreutils
+  core/curl
+  core/glibc
+  core/openjdk11
+  core/openssl
+  core/procps-ng
+  core/ruby
+  core/wget
+  core/zlib
+)
+
+pkg_bin_dirs=(bin)
+pkg_binds_optional=(
+  [elasticsearch]="http-port transport-port"
+)
+pkg_lib_dirs=(lib)
+pkg_exports=(
+  [http-port]=es_yaml.http.port
+  [transport-port]=es_yaml.transport.tcp.port
+)
+pkg_exposes=(http-port transport-port)
+
+OPENDISTRO_DEPENDENCIES=(
+  'security-parent'
+  'security-ssl'
+  'security-advanced-modules'
+)
+
+do_download() {
+  wget -O "${HAB_CACHE_SRC_PATH}/elasticsearch-oss-${ELASTICSEARCH_VERSION}.tar.gz" "${ELASTICSEARCH_PKG_URL}"
```

Comment:
We should use `download_file` here instead of wget, to consolidate download functionality, and to add SHASUM checking for the downloaded file to ensure integrity. See https://github.com/habitat-sh/habitat/blob/master/components/plan-build/bin/public.bash#L350

Reply:
Done! I had no idea about download_file, and now I'll use it <3